### PR TITLE
Moving `Relation.Binary.X` to `Data.X.Relation`

### PR DESCRIPTION
**Pull request by MatthewDaggitt**
As discussed in #173 I've moved the relations for data `X` from `Relation.Binary.X` to `Data.X.Relations`. The existing files re-export the definitions publicly from their new locations to preserve backwards compatibility.

The only two names I'm not quite happy with are:
- `Relation.Binary.Sigma.Pointwise`      ↦ `Data.Product.Relations.DependantPointwise`
- `Relation.Binary.Sum`                  ↦ `Data.Sum.Relations.General`
If anyone has any better ideas for these names feel free to suggest them? I am planning to split the `Pointwise` definition out of `Data.Sum.Relations.General` later after which the naming of `General` may make more sense.

### CODE REVIEW

**Comment by MatthewDaggitt** [comment]
We will of course need to figure out what to do about the now more obviously duplicated `Data.Vec.All.All2` and `Data.Vec.Relations.Pointwise`....

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,21 @@ Deprecated features
 Deprecated features still exist and therefore existing code should still work
 but they may be removed in some future release of the library.
 
+* Relations over data have been moved from the `Relation` subtree to the `Data`
+  subtree. The full list of moves is as follows:
+  - `Relation.Binary.List.Pointwise`       ↦ `Data.List.Relations.Pointwise`
```
Comment (andreasabel) [inline CHANGELOG.md]: Minor bikeshedding:  Why not `Relation` instead of `Relations`?  I think in general we use singular rather than plural.

**Comment by andreasabel** [comment]
Instead of `DependentPointwise` we could also use the name `SigmaPointwise`.

**Comment by MatthewDaggitt** [comment]
Thanks for the feedback @andreasabel I agree with your two suggestions and implemented them.

**Comment by andreasabel** [comment]
Thanks @MatthewDaggitt for this refactoring!

**Pull request merged**